chi/network.py

```diff
@@ -703,34 +712,47 @@ def nuke_network(network_name):
     Args:
         network_name (str): The network name.
     """
+    network = get_network(network_name)
     network_id = get_network_id(network_name)
-
-    # Detach network's subnets from router
-    port = next(iter([
-        p for p in list_ports()
-        if (p['device_owner'] == 'network:router_interface' and
-            p['network_id'] == network_id)
-    ]), None)
-    router_id = port['device_id'] if port else None
-    if router_id:
-        for fixed_ip in port['fixed_ips']:
-            subnet_id = fixed_ip['subnet_id']
-            remove_subnet_from_router(router_id, fixed_ip['subnet_id'])
-            print(f'Detached subnet {subnet_id}')
-        delete_router(router_id)
-        print(f'Deleted router {router_id}')
-
-    # TODO: also detach any running instances (?)
-
-    # Delete all subnets
-    for subnet in list_subnets():
+    print('next network')
+    print(json.dumps(network, indent=2))
+    
+    #Detach the router from all of its networks
+    router_device_id=None
+    for port in neutron().list_ports()['ports']:
+        if port['device_owner'] == "network:router_interface" and port['network_id'] == network_id:
+            router_device_id = port['device_id']
+            print('next port: router_device_id' + router_device_id)
+            print(json.dumps(port, indent=2))
+            break
+        port=None
+    if port != None:
```

Comment:
Using this `port` variable from the prior loop is unexpected. It would be pretty easy to break this in the future, by either removing the `port = None` or any other number of ways. Suggestion: store all such ports in a list (there could be more than 1, I think. At least, w/ the structure, it is possible.)

```python
router_ports = [
    port for port in neutron().list_ports()["ports"]
    if port["device_owner"] == "network:router_interface" and port["network_id"] == network_id
]
```

Looking through the rest of the logic, I don't understand why much of it is needed. `remove_subnet_from_router` can operate solely on information already stored in the ports, it would seem. So you can do

```python
for port in router_ports:
    for fixed_ip in port["fixed_ips"]:
        router_id = port["device_id"]
        remove_subnet_from_router(router_id, fixed_ip["subnet_id"])
```

Done!